drivers/i2c/i2c_sam0.c

```diff
@@ -60,6 +85,50 @@ struct i2c_sam0_dev_data {
 	uint8_t num_msgs;
 };
 
+static int i2c_sam0_recover_bus(const struct device *dev)
+{
+	const struct i2c_sam0_dev_config *cfg = dev->config;
+	pinctrl_soc_pin_t soc_pin_sda, soc_pin_scl;
+
+	/* Save original pin config values */
+	soc_pin_sda = cfg->pcfg->states[PINCTRL_STATE_DEFAULT].pins[SERCOM_I2CM_SDA_PIN_IDX];
+	soc_pin_scl = cfg->pcfg->states[PINCTRL_STATE_DEFAULT].pins[SERCOM_I2CM_SCL_PIN_IDX];
+
+	uint8_t sda_port_idx = SAM_PINMUX_PORT_GET(soc_pin_sda);
+	uint8_t scl_port_idx = SAM_PINMUX_PORT_GET(soc_pin_scl);
+	uint32_t sda_pin = SAM_PINMUX_PIN_GET(soc_pin_sda);
+	uint32_t scl_pin = SAM_PINMUX_PIN_GET(soc_pin_scl);
+	const struct device *sda_dev = sam_port_devs[sda_port_idx];
+	const struct device *scl_dev = sam_port_devs[scl_port_idx];
```

Comment:
Could you pointing out an example where you found different ports for I2C pins at same?